stdlib/src/collections/dict.mojo

```diff
@@ -444,6 +444,26 @@ struct Dict[K: KeyElement, V: CollectionElement](
         self._entries = Self._new_entries(Self._initial_reservation)
         self._index = _DictIndex(len(self._entries))
 
+    @always_inline
+    fn __init__(inout self, *, power_of_two_initial_capacity: Int):
+        """Initialize an empty dictiontary with a pre-reserved initial capacity.
+
+        Args:
+            power_of_two_initial_capacity: At least 8, has to be a power of two.
```

Comment:
**Suggestion** Add a `debug_assert` to check this invariant.  Another approach would be to round up the initial capacity provided to the nearest power of two to retain the class invariants.

Reply:
Yes, commit [f7c4d94](https://github.com/modularml/mojo/pull/3171/commits/f7c4d9498c8b8c4ec34312f1ab5432c816e04387) brings a `debug_assert` :+1: